src/openforms/submissions/tasks.py

```diff
@@ -0,0 +1,232 @@
+            output_field=IntegerField(),
+        ),
+        time_since_creation=ExpressionWrapper(
+            Value(timezone.now(), DateTimeField()) - F("created_on"),
+            output_field=DurationField(),
+        ),
+    ).filter(
+        registration_status=RegistrationStatuses.success,
+        removal_method=RemovalMethods.make_anonymous,
+        time_since_creation__gt=(timedelta(days=1) * F("removal_limit")),
+        _is_cleaned=False,
+    )
+
+    incomplete_submissions = Submission.objects.annotate(
+        removal_method=Case(
+            When(
+                form__incomplete_submissions_removal_method=None,
+                then=Value(config.incomplete_submissions_removal_method),
+            ),
+            default=F("form__incomplete_submissions_removal_method"),
+            output_field=CharField(),
+        ),
+        removal_limit=Case(
+            When(
+                form__incomplete_submissions_removal_limit=None,
+                then=Value(config.incomplete_submissions_removal_limit),
+            ),
+            default=F("form__incomplete_submissions_removal_limit"),
+            output_field=IntegerField(),
+        ),
+        time_since_creation=ExpressionWrapper(
+            Value(timezone.now(), DateTimeField()) - F("created_on"),
+            output_field=DurationField(),
+        ),
+    ).filter(
+        registration_status__in=[
+            RegistrationStatuses.pending,
+            RegistrationStatuses.in_progress,
+        ],
+        removal_method=RemovalMethods.make_anonymous,
+        time_since_creation__gt=(timedelta(days=1) * F("removal_limit")),
+        _is_cleaned=False,
+    )
+
+    errored_submissions = Submission.objects.annotate(
+        removal_method=Case(
+            When(
+                form__errored_submissions_removal_method=None,
+                then=Value(config.errored_submissions_removal_method),
+            ),
+            default=F("form__errored_submissions_removal_method"),
+            output_field=CharField(),
+        ),
+        removal_limit=Case(
+            When(
+                form__errored_submissions_removal_limit=None,
+                then=Value(config.errored_submissions_removal_limit),
+            ),
+            default=F("form__errored_submissions_removal_limit"),
+            output_field=IntegerField(),
+        ),
+        time_since_creation=ExpressionWrapper(
+            Value(timezone.now(), DateTimeField()) - F("created_on"),
+            output_field=DurationField(),
+        ),
+    ).filter(
+        registration_status=RegistrationStatuses.failed,
+        removal_method=RemovalMethods.make_anonymous,
+        time_since_creation__gt=(timedelta(days=1) * F("removal_limit")),
+        _is_cleaned=False,
+    )
+
+    for submission in (
+        list(successful_submissions)
+        + list(incomplete_submissions)
+        + list(errored_submissions)
+    ):
```

Comment:
this potentially loads a lot of data in memory, I'd rather use a solution using `itertools.chain`:

```suggestion
    for submission in itertools.chain(
        successful_submissions.iterator(), 
        incomplete_submissions.iterator(), 
        errored_submissions.iterator()
    ):
```

`queryset.iterator()` has an extremely low memory footprint since it doesn't cache the queryset results. I think it even uses server side database cursors, but that I'm not sure.